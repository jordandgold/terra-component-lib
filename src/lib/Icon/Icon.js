import React from "react";
import PropTypes from "prop-types";

const url = require("./svg/icons.svg");
import "./Icon.scss";

const Icon = props => (
  <svg
    viewBox="0 0 8 8"
    className={`ter-icon ter-icon--${props.type} ter-icon--${props.size} ${
      props.className
    }`}
  >
    <use xlinkHref={`${url}#icon-${props.type}`} />
  </svg>
);

export default Icon;

Icon.propTypes = {
  type: PropTypes.string.isRequired,
  size: PropTypes.string.isRequired,
  className: PropTypes.string
};
