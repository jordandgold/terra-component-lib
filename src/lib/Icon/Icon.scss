@import "../global-styles/terra.scss";

.ter-icon {
  width: 16px;
  height: 16px;
  &--8px {
    width: 8px;
    height: 8px;
  }
}
