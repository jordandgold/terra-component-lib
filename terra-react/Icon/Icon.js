import React from "react";

var url = require("./svg/icons.svg");

import "./Icon.scss";

var Icon = function Icon(props) {
  return React.createElement("svg", {
    viewBox: "0 0 8 8",
    className: "ter-icon ter-icon--".concat(props.type, " ter-icon--").concat(props.size, " ").concat(props.className)
  }, React.createElement("use", {
    xlinkHref: "".concat(url, "#ter-icon--").concat(props.type)
  }));
};

export default Icon;