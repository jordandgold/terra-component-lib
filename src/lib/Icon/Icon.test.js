import React from "react";
import { shallow } from "enzyme";
import Icon from "./Icon";

describe("Iron", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Icon type="open-caret-right-dark-8px" size="8px" />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
