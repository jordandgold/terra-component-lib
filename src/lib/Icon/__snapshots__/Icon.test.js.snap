// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Iron should match the snapshot 1`] = `
<svg
  className="ter-icon ter-icon--open-caret-right-dark-8px ter-icon--8px"
  viewBox="0 0 8 8"
>
  <use
    xlinkHref="[object Object]#icon-open-caret-right-dark-8px"
  />
</svg>
`;
